routers/repo/pull_review.go

```diff
@@ -58,11 +87,32 @@ func CreateCodeComment(ctx *context.Context, form auth.CodeCommentForm) {
 	}
 
 	log.Trace("Comment created: %-v #%d[%d] Comment[%d]", ctx.Repo.Repository, issue.Index, issue.ID, comment.ID)
+
+	if form.Origin == "diff" {
+		comments, err := models.FetchCodeCommentsByLine(issue, ctx.User, comment.TreePath, comment.Line)
+		if err != nil {
+			ctx.ServerError("FetchCodeCommentsByLine", err)
+			return
+		}
+		ctx.Data["PageIsPullFiles"] = true
+		ctx.Data["comments"] = comments
+		ctx.Data["CanMarkConversation"] = true
+		ctx.Data["Issue"] = issue
+		pullHeadCommitID, err := ctx.Repo.GitRepo.GetRefCommitID(issue.PullRequest.GetGitRefName())
+		if err != nil {
+			ctx.ServerError("GetRefCommitID", err)
+			return
+		}
+		ctx.Data["AfterCommitID"] = pullHeadCommitID
+		ctx.HTML(200, tplConversation)
+		return
+	}
```

Comment:
This block is mostly duplicate with the block below. Maybe it's worth factoring out the duplicated parts into a separate function?